src/Cli/dotnet/NugetSearch/NugetToolSearchApiRequest.cs

```diff
@@ -77,5 +79,14 @@ public async Task<string> GetResult(NugetSearchApiParameter nugetSearchApiParame
 
             return uriBuilder.Uri;
         }
+
+        // More detail on this API https://github.com/dotnet/sdk/issues/12038
+        private static async Task<Uri> DomainAndPath()
+        {
+            var repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+            var resource = await repository.GetResourceAsync<ServiceIndexResourceV3>();
+            var uris = resource.GetServiceEntryUris("SearchQueryService");
```

Comment:
Please use `SearchQueryService/3.5.0`. This marks the service entry that supports package type filtering (which is required by your scenario).

Reply:
You could even link to here in your code comment:
https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource#versioning